In [3]:
import pandas as pd
import numpy as np
%matplotlib inline
import matplotlib.pyplot as plt


In [33]:
Air = pd.read_csv("AQI_Data.csv")

In [34]:
Air.head(10)

date               site parameter  index_value description  \
0  2016-01-01      Lawrenceville     PM25B           25        Good   
1  2016-01-01         Flag Plaza        CO            0        Good   
2  2016-01-01  Harrison Township     OZONE           31        Good   
3  2016-01-01             Avalon       SO2           10        Good   
4  2016-01-01            Lincoln      PM25           35        Good   
5  2016-01-01            Lincoln      PM10           11        Good   
6  2016-01-01      South Fayette       SO2            3        Good   
7  2016-01-01     North Braddock     PM10B           12        Good   
8  2016-01-01       Parkway East        CO            4        Good   
9  2016-01-01         Flag Plaza      PM10            7        Good   

  health_advisory health_effects  
0             NaN            NaN  
1             NaN            NaN  
2             NaN            NaN  
3             NaN            NaN  
4             NaN            NaN  
5             NaN            NaN  
6             NaN            NaN  
7             NaN            NaN  
8             NaN            NaN  
9             NaN            NaN

0-50 index is considered Good, 50 - 100 is considered Moderate, 100-130? is considered Unhealthy for sensitive groups

In [35]:
Air['index_value'].mean()

21.731824565193907

The average Air Quality Index Value for all the neighborhoods/sites in Allegheny County is 21.731824565193907.
Anything below that index value is considered better than average air quality relative to the other neigborhoods/sitesin Allegheny County

In [54]:
n = 0
uniq = Air['site'].unique()
for a in uniq:
    n = n + 1
n = str(n)
print("There are " + n + " unique neighborhoods/sites that were recorded")

There are 15 unique neighborhoods/sites that were recorded


We are gonna average the air quality index for each neighborhood/site and find the lowest average Air Quality Index, and crown that as that healthiest neigborhood to live in.

First, we're gonna find out how many times each neighborhood/site was recorded

In [53]:
AQI = {}
n=0
n1=0
n2=1
for row in Air.iterrows():
    n1 = Air.iloc[n,1] 
    if n1 not in AQI:
        AQI[n1] = n2
    elif n1 in AQI:
        AQI[n1] = AQI[n1] + n2
    n = n+1

AQI

{'Lawrenceville': 4026,
 'Flag Plaza': 4002,
 'Harrison Township': 4806,
 'Avalon': 4054,
 'Lincoln': 3574,
 'South Fayette': 4627,
 'North Braddock': 5763,
 'Parkway East': 7460,
 'Liberty 2': 4906,
 'Lawrenceville 2': 4970,
 'Glassport High Street': 2493,
 'Liberty': 2506,
 'Clairton': 553,
 'West Mifflin': 301,
 'Pittsburgh': 1501}

The most recorded neighborhood/site is Parkway East, while the least recorded neigborhood/site is West Mifflin

Now, we are gonna add the Air Quality index for each neighborhood/site, then divide by how many times that neighborhood/site has been recorded so we can find the average Air Quality index for said neighborhood/site